registry/client.go

```diff
@@ -352,3 +352,118 @@ func (e Error) Error() string {
 
 	return "registry error: " + strconv.Itoa(e.StatusCode)
 }
+
+// Compatibility levels.
+const (
+	BackwardCL           string = "BACKWARD"
+	BackwardTransitiveCL string = "BACKWARD_TRANSITIVE"
+	ForwardCL            string = "FORWARD"
+	ForwardTransitiveCL  string = "FORWARD_TRANSITIVE"
+	FullCL               string = "FULL"
+	FullTransitiveCL     string = "FULL_TRANSITIVE"
+	NoneCL               string = "NONE"
+)
+
+/*
+this function returns whether or not a compatibility level
+is valid according to the ones described in:
+https://docs.confluent.io/platform/current/schema-registry/develop/api.html#compatibility
+.
+*/
+func validCompatibilityLevel(compatibilityLevel string) bool {
+	switch compatibilityLevel {
+	case BackwardCL:
+		return true
+	case BackwardTransitiveCL:
+		return true
+	case ForwardCL:
+		return true
+	case ForwardTransitiveCL:
+		return true
+	case FullCL:
+		return true
+	case FullTransitiveCL:
+		return true
+	case NoneCL:
+		return true
+	default:
+		return false
+	}
+}
+
+// compatibilityPayload is the response body obtained from the schema registry when updating a compatibility level.
+type compatibilityPayload struct {
+	Compatibility string `json:"compatibility"`
+}
+
+/*
+errByInvalidCL returns error for and invalid compatibility level, i.e. a compatibility
+level not present in:
+https://docs.confluent.io/platform/current/schema-registry/develop/api.html#compatibility .
+*/
+func errByInvalidCL(compatibilityLevel string) error {
+	return fmt.Errorf("invalid compatibility level %s", compatibilityLevel)
+}
+
+// PutCompatibilityLevelGlobal sets the global compatibility level of the registry.
+func (c *Client) PutCompatibilityLevelGlobal(
+	ctx context.Context,
+	compatibilityLevel string,
+) (string, error) {
```

Comment:
I am confused by the returned compatibility level (`string`) in this function and the next? Either it worked and it is the compatibility level you set, or it didn't and it would return an error. What is the need here?

Reply:
The API returns the compatibility level that was set according to the [docs](GetCompatibilityLevelGlobal). So boh, I thought of returning it that well (shrug)

Reply:
Regarding this point, personally for me it's the same.

I would say that a method returning the compatibility level could more "related" to the actual API, while a mehod not returning that string would make maybe more sense from the usage point of view.

I'm open to every decision you make regarding this

Reply:
In general I think it would lead to a lot of code ignoring the return, and I cannot think of a single use cause where the return would be used. Personally I would not return it.

Reply:
done